tardis/montecarlo/montecarlo_numba/tests/test_cuda_formal_integral.py

```diff
@@ -0,0 +1,468 @@
+
+    cuda_searchsorted_value_right_caller[1, 1](line_list_nu, nu_insert, actual)
+    
+    ntest.assert_almost_equal(actual, expected)
+
+@cuda.jit
+def cuda_searchsorted_value_right_caller(line_list_nu, nu_insert, actual):
+    x = cuda.grid(1)
+    actual[x] = len(line_list_nu) - 1 - cuda_searchsorted_value_right(line_list_nu[::-1], nu_insert)
+
+
+    
+    
+ 
+from tardis.montecarlo.montecarlo_numba.formal_integral import FormalIntegrator
+from tardis.montecarlo.montecarlo_numba.formal_integral_cuda_test import FormalIntegrator as cuda_FormalIntegrator
+
+from tardis.io.atom_data.util import download_atom_data
+from tardis import run_tardis
+from astropy import units as u
+from tardis.io.config_reader import Configuration
+from tardis.simulation import Simulation
+
+from tardis.montecarlo.montecarlo_numba.r_packet import (RPacket)
+from tardis.montecarlo.montecarlo_numba.numba_interface import (NumbaModel, 
+                                                                NumbaPlasma, 
+                                                                numba_plasma_initialize, 
+                                                                Estimators)
+
+from basic_types_Copy1 import (get_numba_model,
+                         get_r_packet,
+                         get_numba_plasma,
+                         get_estimator) #This file is in the same test directory for easy access right now. 
+
+from tardis.montecarlo import MontecarloRunner
+from tardis.model import Radial1DModel
+from tardis.plasma.standard_plasmas import assemble_plasma
+from tardis.io.util import HDFWriterMixin
+from tardis.io.config_reader import ConfigurationError
+
+
+#This test is a mess right now, I am just trying to see how to start it
+@pytest.mark.parametrize(
+    ["no_of_packets", "iterations"],
+    [
+        (40000, 1),
+    ],
+)
+def test_full_formal_integral(no_of_packets, iterations):
+    """
+    This tests the full formal integral
+    """
+    download_atom_data('kurucz_cd23_chianti_H_He')
+    #This file is a simple file, 1 iteration with only 40000 packets.
+    tardis_config = Configuration.from_yaml('tardis/montecarlo/montecarlo_numba/tests/tardis_example_verysimple.yml') 
+    #current throws an error here.
+    """
+    OSError: Atom Data tardis/montecarlo/montecarlo_numba/tests/kurucz_cd23_chianti_H_He.h5 is not found in current path or in TARDIS data repo. tardis/montecarlo/montecarlo_numba/tests/kurucz_cd23_chianti_H_He is also not a standard known TARDIS atom dataset.
+
+tardis/io/atom_data/util.py:48: OSError
+    """
+    sim1 = Simulation.from_config(tardis_config)
+    sim1.run()
+    
+    basic_estimator1 = get_estimator(sim1)
+    basic_estimator2 = get_estimator(sim1)
+    
+    tau_sobolev_shape1 = basic_estimator1.Edotlu_estimator.shape
+    tau_sobolev_shape2 = basic_estimator2.Edotlu_estimator.shape
+    
+    basic_runner1 = MontecarloRunner.from_config(tardis_config, None, False)
```

Comment:
You only need a single runner.  In fact, you *should* use a single runner to guarantee the formal integral is acting on the same data